t/deadlock.t

```diff
@@ -0,0 +1,153 @@
+# vim:set ft= ts=4 sw=4 et fdm=marker:
+use Test::Nginx::Socket::Lua;
+
+master_on();
+
+plan tests => repeat_each() * (blocks() * 17);
+
+$ENV{TEST_NGINX_HTML_DIR} ||= html_dir();
+
+master_on();
+workers(3);
+run_tests();
+
+__DATA__
+
+=== TEST 1: posting events and handling events, broadcast
+--- http_config
+    lua_package_path "../lua-resty-core/lib/?.lua;lualib/?/init.lua;lualib/?.lua;;";
+    lua_shared_dict dict 1m;
+    init_by_lua_block {
+        require("ngx.process").enable_privileged_agent(4096)
+    }
+    init_worker_by_lua_block {
+        local opts = {
+            broker_id = 2,
+            listening = "unix:$TEST_NGINX_HTML_DIR/nginx.sock",
+        }
+
+        local ev = require("resty.events").new(opts)
+        if not ev then
+            ngx.log(ngx.ERR, "failed to new events")
+        end
+
+        local ok, err = ev:init_worker()
+        if not ok then
+            ngx.log(ngx.ERR, "failed to init_worker events: ", err)
+        end
+
+        local semaphore = require("ngx.semaphore")
+        local sema = semaphore.new(1)
+
+        local id = ngx.worker.id()
+
+        if id == 2 then
+            ngx.shared.dict:set("broker-pid", ngx.worker.pid())
+        end
+
+        if id == 0 or id == 1 then
+            ev:subscribe("content_by_lua", "first", function()
+                ngx.sleep(0)
+                ngx.log(ngx.INFO, id, " first handler")
+                local ok, err = sema:wait(0)
+                if ok then
+                    ngx.log(ngx.INFO, id, " first got mutex")
+                    ngx.sleep(5)
+                else
+                    ngx.log(ngx.CRIT, id, " first did not got mutex: ", err)
+                end
+                sema:post()
+            end)
+        end
+
+        if id == nil then
+            ev:subscribe("content_by_lua", "second", function()
+                ngx.sleep(0)
+                ngx.log(ngx.INFO, "privileged killer")
+                ngx.timer.at(0, function()
+                    ngx.log(ngx.INFO, "killing ", ngx.shared.dict:get("broker-pid"))
+                    os.execute("kill -9 " .. ngx.shared.dict:get("broker-pid"))
```

Comment:
According to my previous observation, we can add a `ngx.sleep(3)` to let the signal from `os.execute` captured.